### PR TITLE
feat(eap): Endpoint to get the tags available for a project

### PR DESCRIPTION
**Pull request by colin-sentry**
This is important for things like the search bar, where you want to discover tags you can use

### CODE REVIEW

File: snuba/web/rpc/tags_list.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from snuba.clickhouse.formatter.nodes import FormattedQuery, StringNode
+from snuba.datasets.plans.storage_processing import get_query_data_source
+from snuba.datasets.storages.factory import get_storage
+from snuba.datasets.storages.storage_key import StorageKey
+from snuba.utils.metrics.timer import Timer
+from snuba.web.rpc.exceptions import BadSnubaRPCRequestException
+
+
+def tags_list_query(
+    request: TagsListRequest, _timer: Optional[Timer] = None
```
Comment (volokluev) [inline snuba/web/rpc/tags_list.py:17]: You can remove the timer arg

File: snuba/web/rpc/tags_list.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    query = f"""
+SELECT * FROM (
+    SELECT DISTINCT attr_key, 'str' as type
+    FROM {str_from.table_name}
+    UNION ALL
+    SELECT DISTINCT attr_key, 'num' as type
+    FROM {num_from.table_name}
+) LIMIT {request.limit} OFFSET {request.offset}
+SETTINGS max_bytes_to_read=1000000000
```
Comment (volokluev) [inline snuba/web/rpc/tags_list.py:49]: remove this, we should just add allocation policies, you can do so in a separate PR

**Review by volokluev — COMMENTED** [review]
Add a test please

File: snuba/web/rpc/tags_list.py (hunk 1 above)
Comment (colin-sentry) [reply snuba/web/rpc/tags_list.py:18]: for the mypy typing on the flask side

**Review by volokluev — APPROVED** [review]
